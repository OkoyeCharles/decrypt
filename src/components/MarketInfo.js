import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchMarketInfo } from '../redux/actions';
import { currencyFormatter, numberFormat } from '../utils';
import '../styles/MarketInfo.css';

function MarketInfo() {
  const dispatch = useDispatch();
  const marketInfo = useSelector((state) => state.marketInfo);
  const [marketInfoLoaded, setMarketInfoLoaded] = useState(false);

  useEffect(() => {
    if (marketInfoLoaded) return;
    dispatch(fetchMarketInfo());
    setMarketInfoLoaded(true);
  }, []);

  return (
    <div className="marketInfo">
      <h2 className="marketInfo__header">Decrypt Market Information</h2>
      <ul className="marketInfo__container">
        <li className="marketInfo__item">
          <h3>Coins</h3>
          <span>{marketInfo.coinsCount ? numberFormat(marketInfo.coinsCount) : 'Loading...'}</span>
        </li>
        <li className="marketInfo__item">
          <h3>Active markets</h3>
          <span>{marketInfo.marketCount ? numberFormat(marketInfo.marketCount) : 'Loading...'}</span>
        </li>
        <li className="marketInfo__item">
          <h3>Market cap</h3>
          <span>{marketInfo.marketCap ? currencyFormatter.format(marketInfo.marketCap) : 'Loading...'}</span>
        </li>
        <li className="marketInfo__item">
          <h3>Market volume</h3>
          <span>{marketInfo.marketVol ? currencyFormatter.format(marketInfo.marketVol) : 'Loading...'}</span>
        </li>
        <li className="marketInfo__item">
          <h3>Market change (avg)</h3>
          <span>{marketInfo.marketPerChange || 'Loading...'}</span>
        </li>
      </ul>
    </div>
  );
}

export default MarketInfo;
