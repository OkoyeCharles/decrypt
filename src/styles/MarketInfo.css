.marketInfo {
  color: #555;
  padding: 1em 0.25em;
  background: #efba0a;
  width: 100%;
}

.marketInfo__header {
  position: relative;
  margin-bottom: 0.25em;
}

.marketInfo__header::after {
  content: '';
  position: absolute;
  height: 0.125em;
  width: 100%;
  bottom: 0;
  left: 0;
  background: #8e7215;
  border-radius: 0.25em;
  transform-origin: left;
  transform: scaleX(0.5);
}

.marketInfo__container {
  list-style: none;
  padding: 0;
  display: grid;
  gap: 0.05em;
  grid-template-columns: repeat(auto-fill, minmax(10em, 1fr));
}

.marketInfo__item {
  border-radius: 0.2em;
  box-shadow: inset 0 0 0 0.01em;
  padding: 0.25em 0.5em;
  min-height: 5em;
  background: white;
}

.marketInfo__item :nth-child(1) {
  border-bottom: 0.1em solid #b8900d86;
}
