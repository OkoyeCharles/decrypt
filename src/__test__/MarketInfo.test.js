import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import MarketInfo from '../components/MarketInfo';

it('<MarketInfo /> renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <MemoryRouter>
          <MarketInfo />
        </MemoryRouter>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
