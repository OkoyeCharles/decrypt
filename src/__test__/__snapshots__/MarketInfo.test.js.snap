// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MarketInfo /> renders correctly 1`] = `
<div
  className="marketInfo"
>
  <h2
    className="marketInfo__header"
  >
    Decrypt Market Information
  </h2>
  <ul
    className="marketInfo__container"
  >
    <li
      className="marketInfo__item"
    >
      <h3>
        Coins
      </h3>
      <span>
        Loading...
      </span>
    </li>
    <li
      className="marketInfo__item"
    >
      <h3>
        Active markets
      </h3>
      <span>
        Loading...
      </span>
    </li>
    <li
      className="marketInfo__item"
    >
      <h3>
        Market cap
      </h3>
      <span>
        Loading...
      </span>
    </li>
    <li
      className="marketInfo__item"
    >
      <h3>
        Market volume
      </h3>
      <span>
        Loading...
      </span>
    </li>
    <li
      className="marketInfo__item"
    >
      <h3>
        Market change (avg)
      </h3>
      <span>
        Loading...
      </span>
    </li>
  </ul>
</div>
`;
